--- ./setup.py.old	2019-01-04 06:07:59.000000000 +0200
+++ ./setup.py		2019-01-16 18:39:39.307046059 +0200
@@ -551,36 +551,35 @@
 def copy_man_pages(ddir):
     mandir = os.path.join(ddir, 'share', 'man')
     safe_makedirs(mandir)
-    try:
+('''    try:
         shutil.rmtree(os.path.join(mandir, 'man1'))
     except FileNotFoundError:
         pass
     src = os.path.join(base, 'docs/_build/man')
     if not os.path.exists(src):
-        raise SystemExit('''\
+        raise SystemExit
 The kitty man page is missing. If you are building from git then run:
 make && make docs
 (needs the sphinx documentation system to be installed)
-''')
     shutil.copytree(src, os.path.join(mandir, 'man1'))
-
+''')
 
 def copy_html_docs(ddir):
     htmldir = os.path.join(ddir, 'share', 'doc', appname, 'html')
     safe_makedirs(os.path.dirname(htmldir))
-    try:
+('''    try:
         shutil.rmtree(htmldir)
     except FileNotFoundError:
         pass
     src = os.path.join(base, 'docs/_build/html')
     if not os.path.exists(src):
-        raise SystemExit('''\
+        raise SystemExit
 The kitty html docs are missing. If you are building from git then run:
 make && make docs
 (needs the sphinx documentation system to be installed)
-''')
-    shutil.copytree(src, htmldir)
 
+    shutil.copytree(src, htmldir)
+''')
 
 def compile_python(base_path):
     import compileall
